packages/tse-backend/src/modules/exchange-data/exchange-data.service.ts

```diff
@@ -0,0 +1,298 @@
+      [exchange: string]: { [symbol: string]: TickerPriceResponse };
+    } = {};
+
+    const exchangeNamesArray = exchangeNames.split(',');
+    const symbolsArray = symbols.split(',');
+
+    exchangeNamesArray.forEach((exchangeName) => {
+      symbolsArray.forEach((symbol) => {
+        const tpCommand = new GetTickerPriceCommand(exchangeName, symbol);
+        const fetchPromise = this.getTickerPrice(tpCommand)
+          .then((price) => {
+            if (!results[exchangeName]) {
+              results[exchangeName] = {};
+            }
+            results[exchangeName][symbol] = price;
+            return price;
+          })
+          .catch((error) => {
+            this.logger.error(
+              `Error fetching ticker price for ${exchangeName}:${symbol}: ${error}`,
+            );
+            return null;
+          });
+        fetchPromises.push(fetchPromise);
+      });
+    });
+
+    await Promise.all(fetchPromises);
+    return results;
+  }
+
+  async getSupportedSymbols(command: GetSupportedSymbolsCommand) {
+    const { exchange } = command;
+    const exchangeInstance = this.exchangeRegistryService.getExchange(exchange);
+    if (!exchangeInstance) {
+      throw new Error(`Exchange ${exchange} is not configured.`);
+    }
+    this.logger.log(`Fetching supported symbols from ${exchangeInstance.name}`);
+
+    try {
+      await exchangeInstance.loadMarkets();
+      return Object.keys(exchangeInstance.markets);
+    } catch (error) {
+      this.logger.error(error);
+      throw error;
+    }
+  }
+
+  private async watchMarketData(
+    type: MarketDataType,
+    exchange: string,
+    symbol: string | undefined,
+    symbols: string[] | undefined,
+    timeFrame: string | undefined,
+    onData: (data: any) => void,
+    extraParams: any = {},
+  ): Promise<void> {
+    const exchangeInstance = this.exchangeRegistryService.getExchange(exchange);
+    const methodName = `watch${type}`;
+    const compositeKey = createCompositeKey(
+      type,
+      exchange,
+      symbol,
+      symbols,
+      timeFrame,
+    );
+
+    if (!exchangeInstance || !exchangeInstance.has[methodName]) {
+      throw new Error(
+        `Exchange ${exchange} does not support ${methodName} or is not configured.`,
+      );
+    }
+
+    try {
+      while (this.subscriptionManager.isSubscribed(compositeKey)) {
+        const data = await exchangeInstance[methodName](
+          symbol || symbols,
+          ...Object.values(extraParams),
+        );
+        onData(data);
+      }
+    } catch (error) {
+      this.logger.error(
+        `Error watching ${type} for ${symbol || symbols} on ${exchange}: ${error.message}`,
+      );
+      await new Promise((resolve) => setTimeout(resolve, 1000)); // Reconnect after a delay
+    }
+  }
+
+  async watchOrderBook(
+    exchange: string,
+    symbol: string,
+    onData: (data: any) => void,
+    limit = 14,
+  ): Promise<void> {
+    if (exchange === 'bitfinex') {
+      limit = 25;
+    }
```

Comment:
Why do we need this?

Reply:
Bitfinex requires the watchOrderBook limit parameter to be set to either undefined, 25, or 100, as part of their API constraints